### PR TITLE
Upgrade openssl to version 1.1.1m

### PR DESCRIPTION
**Pull request by xyzhao2018**
Signed-off-by: xzhao3 <shirley.zhao@intel.com>

### CODE REVIEW

**Review by GangSecurity — APPROVED** [review]

**Comment by GangSecurity** [comment]
Is it possible to update libcrypto.a ony, not update the headfiles?

**Review by jingdlu — COMMENTED** [review]
Did you generate libcrypto.a with -fPIC?

**Review by jingdlu — COMMENTED** [review]
You are changing so many common files that impact Arm. @kwang13 Is that OK?

File: host/xtest/Makefile [hunk 1]
```diff
@@ -44,9 +44,13 @@ ifneq ($(OPTEE_OPENSSL_EXPORT),)
...
 CFLAGS += -I$(OPTEE_OPENSSL_EXPORT)
 else #OPTEE_OPENSSL_EXPORT
+ifeq ($(ARCH),x86_64)
+CFLAGS += -I../openssl_1.1.1m/include
+else
 CFLAGS += -I../openssl/include
+endif
 ifeq ($(ARCH),x86_64)
-LDFLAGS += ../openssl/lib/x86_64/libcrypto.a -ldl
+LDFLAGS += ../openssl_1.1.1m/lib/x86_64/libcrypto.a -ldl
```
Comment (jingdlu) [inline host/xtest/Makefile:49]: Can you put CFLAGs and LDFLAGs changes for x86 to one place?

**Review by jingdlu — APPROVED** [review]

**Pull request merged**